gui/video_thread.py

```diff
@@ -122,8 +123,20 @@ def on_select_filenames(self, filenames):
                 if json_data["sources"]:
                     for source in json_data["sources"]:
                         api = cv2.CAP_FFMPEG
-                        if source["api"] == "gstreamer":
+
+                        # gstreamer or gstreamer-pipeline: [name] is the full gstreamer pipeline
+                        if source["api"] == "gstreamer" or source["api"] == "gstreamer-pipeline":
+                            api = cv2.CAP_GSTREAMER
+                        # gstreamer-record: [name] is the port (i.e. 5600) to be both streamed and displayed
+                        elif source["api"] == "gstreamer-record":
+                            api = cv2.CAP_GSTREAMER
+                            recording_file = datetime.datetime.now().strftime("recordings/%Y-%m-%d_%H%M%S.flv")
+                            source["name"] = "udpsrc port=" + source["name"] + " caps=\"application/x-rtp, media=(string)video, clock-rate=(int)90000, encoding-name=(string)H264, payload=(int)96\" ! rtph264depay ! h264parse ! tee name=t t. ! queue ! flvmux ! filesink location=" + recording_file + " t. ! queue leaky=1 ! decodebin ! videoconvert ! appsink"

```

Comment:
It might be better to have the gstreamer pipeline strings in a separate config file. We should be switching these over to mjpeg encoding soon.